Rubberduck.Parsing/VBA/DeclarationResolveRunner.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Parsing.VBA
+{
+    public class DeclarationResolveRunner : DeclarationResolveRunnerBase
+    {
+        private const int _maxDegreeOfDeclarationResolverParallelism = -1;
+
+        public DeclarationResolveRunner(
+            RubberduckParserState state, 
+            IParserStateManager parserStateManager, 
+            ICOMReferenceManager comReferenceManager) 
+        :base(
+             state, 
+             parserStateManager, 
+             comReferenceManager)
+        { }
+
+        public override void ResolveDeclarations(ICollection<QualifiedModuleName> modules, CancellationToken token)
+        {
+            if (!modules.Any())
+            {
+                return;
+            }
+
+            _projectDeclarations.Clear();
+
+                token.ThrowIfCancellationRequested();
```

Comment:
Additional note: throwing here really isn't necessary ... 
```
if (token.IsCancellationRequested) 
{
    return;
}
```
is equally effective and doesn't generate exception-noise that we need to flatten in a runner...